### PR TITLE
Fix test result PR comment

### PR DESCRIPTION
**Pull request by joseph-sentry**
This PR adds escaping for certain characters when making the test result PR comment.

We have to do this to properly display the failure message table.

### CODE REVIEW

File: helpers/string.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from typing import List
+
+
+class EscapeEnum(Enum):
+    APPEND = "append"
+    PREPEND = "prepend"
+    REPLACE = "replace"
+
+
+class Replacement:
```
Comment (giovanni-guidini) [inline helpers/string.py:11]: [nit]
I believe you can use a [@dataclass](https://docs.python.org/3/library/dataclasses.html) for this. That way you don't have to specify the `__init__` function, just the fields.

```python
@dataclass
class Replacement:
    chars: str
    output: str
    method: EscapeEnum
```

File: helpers/string.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def __init__(self, chars: str, output: str, method: EscapeEnum):
+        self.chars = chars
+        self.output = output
+        self.method = method
+
+
+class StringEscaper:
+    """
+    Class to use to escape strings using format defined
```
Comment (giovanni-guidini) [inline helpers/string.py:20]: Amazing docs 👏
thanks for that

File: helpers/string.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        to each char in Replacement.chars using the char in output
+
+        for example:
+            escape_def = [
+                Replacement(chars="<>", output="\\", method=EscapeEnum.PREPEND),
+                Replacement(chars="pre", output="div", method=EscapeEnum.REPLACE)
+            ]
+            StringEscaper(escape_def)
+
+            replace("<pre></pre>")
```
Comment (giovanni-guidini) [inline helpers/string.py:37]: If you apply the `replace` function for every _char_ in this `Replacement` definition wouldn't the output actually be `\<divdivdiv\>\</divdivdiv\>` ?

File: helpers/tests/unit/test_string.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from helpers.string import EscapeEnum, Replacement, StringEscaper
+
+
+def test_string_escaper():
```
Comment (giovanni-guidini) [inline helpers/tests/unit/test_string.py:4]: I suggest you include the example in the documentation here

File: helpers/string.py (hunk 3 above)
Comment (joseph-sentry) [reply helpers/string.py:37]: good point, that was a bad example, but this also brings up the point that maybe this should include an option for specifying strings to replace, append, or prepend to

**Review by giovanni-guidini — APPROVED** [review]

**Pull request merged**